### PR TITLE
Adds buff to wearing the cakehat for extended periods of time.

### PR DESCRIPTION
**Pull request by Shadowlight213**
remake of #2509
@Incoming5643 
Having issues with the virus curing while the hat is on and leaving a null in the list and re-applying itself or something.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
I...I don't understand.

File: code/datums/diseases/transformation.dm [hunk 1]
```diff
@@ -228,3 +228,27 @@
...
+	cure_text = "Take off the damn hat"
+	agent = "Dixbfloppin"
+	desc = "This disease transforms the victim into a cake."
+	visibility_flags = 1
+	stage_prob = 1
+	stage1	= list("This cake hat is so bitching...")
+	stage2	= list("You crave frosting.")
+	stage3	= list("<span class='danger'> The cake is all", "\red So fucking bitching</span>")
+	stage4	= list("<span class='danger'> THE CAKE THE CAKE THE CAKE THE CAKE</span>")
+	stage5	= list("<span class='danger'> YOU ARE NOW DELICIOUS</span>")
```
Comment (Miauw62) [inline code/datums/diseases/transformation.dm]: span classes :+1:

File: code/datums/diseases/transformation.dm [hunk 2]
```diff
@@ -228,3 +228,27 @@
...
+/datum/disease/transformation/cake
+	name = "Cake Hat Infection"
+	cure_text = "Take off the damn hat"
+	agent = "Dixbfloppin"
+	desc = "This disease transforms the victim into a cake."
+	visibility_flags = 1
+	stage_prob = 1
+	stage1	= list("This cake hat is so bitching...")
+	stage2	= list("You crave frosting.")
+	stage3	= list("<span class='danger'> The cake is all", "\red So fucking bitching</span>")
```
Comment (octareenroon91) [inline code/datums/diseases/transformation.dm]: Two list entries, each with half a span?

**Comment by vista-** [comment]
Sounds like a fitting and tasty end for shitters who break into bar/kitchen. :+1:

File: code/datums/diseases/transformation.dm (hunk 2 above)
Comment (neersighted) [inline code/datums/diseases/transformation.dm]: Also remove the space please

File: code/modules/clothing/head/misc_special.dm [hunk 3]
```diff
@@ -62,7 +62,10 @@
 		var/mob/living/carbon/human/M = location
 		if(M.l_hand == src || M.r_hand == src || M.head == src)
 			location = M.loc
-
+		for(var/datum/disease/A in M.viruses)
+			if(istype(A, /datum/disease/transformation/cake/))
```
Comment (neersighted) [inline code/modules/clothing/head/misc_special.dm]: I think we have an `istype_in_list` for this.

**Comment by octareenroon91** [comment]
Seems like this has potential for grief that doesn't stand out in the logs.

**Comment by Zilenan91** [comment]
Wow this is certainly a thing

File: code/datums/diseases/transformation.dm [hunk 4]
```diff
@@ -228,3 +228,27 @@
 		if(4)
 			if (prob(20))
 				affected_mob.say(pick("Bark!", "AUUUUUU"))
+
+
+/datum/disease/transformation/cake
+	name = "Cake Hat Infection"
+	cure_text = "Take off the damn hat"
+	agent = "Dixbfloppin"
```
Comment (Incoming5643) [inline code/datums/diseases/transformation.dm]: Transcending history, and the world, a tale of souls and swords, eternally retold.

File: code/datums/diseases/transformation.dm (hunk 2 above)
Comment (Shadowlight213) [inline code/datums/diseases/transformation.dm]: Whoops.

**Comment by Incoming5643** [comment]
:grey_exclamation: :exclamation: I disavow personal culpability for meme code I made years ago :exclamation: :grey_exclamation:

**Comment by Shadowlight213** [comment]
With a stage probability of one, you'd have to wear the hat for a long time to get turned. To force cake someone, you'd need to restrain them which is logged. Taking the hat should instantly cure the virus, so I don't think this will be something that you'd fully experience unwillingly.

File: code/datums/diseases/transformation.dm (hunk 4 above)
Comment (Shadowlight213) [inline code/datums/diseases/transformation.dm]: Goof said he wants to use this as the name for a new reagent.

**Comment by Anonmare** [comment]
>Genetics makes monkey men
>Gives monkey men to chef
>Chef puts hat on monkey men
>???
>PROFIT (and diabetes)

**Comment by PKPenguin321** [comment]
Yesssssss :+1:

**Comment by Zilenan91** [comment]
Does the hat get deleted when they cake or does it drop on the floor

**Comment by Lumipharon** [comment]
I sentence you to death by hat!

**Comment by Incoming5643** [comment]
the hat isn't preserved, this is a single use death sentence item.

It's also way too wacky :hankey:

**Comment by Zilenan91** [comment]
The hat should be preserved so people become consumed by the cake

**Comment by PKPenguin321** [comment]
make a cursed variant of the cakehat that becomes nodrop when it's placed on your head and creates two more of itself when you're converted

**Comment by Shadowlight213** [comment]
I need someone to help me fix the curing and null viruses issue before I can think about expanding this.

**Comment by drynwyn** [comment]
>make a cursed variant of the cakehat that becomes nodrop when it's placed on your head and creates two more of itself when you're converted

I'm going to do this even if this pull gets closed.

**Comment by octareenroon91** [comment]
>>make a cursed variant of the cakehat that becomes nodrop when it's placed on your head and creates two more of itself when you're converted

>I'm going to do this even if this pull gets closed.

Cake mode when?

**Comment by duncathan** [comment]
I really hope nobody is taking this seriously

**Comment by PKPenguin321** [comment]
*i am*

**Comment by Lumipharon** [comment]
This should be a special version of the normal cake hat, for away mission/mining room loot or something. 

pls no close.

**Comment by Iamgoofball** [comment]
what the fuck dunc

**Comment by Shadowlight213** [comment]
@duncathan Rude. I was.

**Comment by drynwyn** [comment]
pls no close, this is the sort of genuinely funny thing we need more of.

**Pull request closed without merging**